symengine/tests/basic/test_series_generic.cpp

```diff
@@ -15,89 +17,135 @@ using SymEngine::Pow;
+
+    RCP<const UnivariateSeries> a = UnivariateSeries::create(x, 5, apoly_);
+    RCP<const UnivariateSeries> b = UnivariateSeries::create(x, 4, bpoly_);
+    RCP<const Basic> c = sub(a, b);
+    RCP<const UnivariateSeries> f = UnivariateSeries::create(x, 4, fpoly_);
+    REQUIRE(c->__cmp__(*f));
+
+    RCP<const UnivariateSeries> d = UnivariateSeries::create(x, 2, bpoly_);
+    RCP<const Basic> e = sub(a, d);
+    RCP<const UnivariateSeries> g = UnivariateSeries::create(x, 2, gpoly_);
+    REQUIRE(e->__cmp__(*f));
+}
+
+TEST_CASE("Multiplication of two UnivariateExprPolynomial with precision", "[UnivariateSeries]")
+{
+    RCP<const Symbol> x  = symbol("x");
+    UnivariateExprPolynomial a(univariate_polynomial(x, 2, {{0, 1}, {1, 2}, {2, 1}}));
+    UnivariateExprPolynomial b(univariate_polynomial(x, 2, {{0, -1}, {1, -2}, {2, -1}}));
+    UnivariateExprPolynomial c(univariate_polynomial(x, 3, {{0, 1}, {1, 4}, {2, 6}, {3, 4}}));
+    UnivariateExprPolynomial d(univariate_polynomial(x, 4, {{0, -1}, {1, -4}, {2, -6}, {3, -4}, {4, -1}}));
+    
+    UnivariateExprPolynomial e = UnivariateSeries::mul(a, a, 4);
+    UnivariateExprPolynomial f = UnivariateSeries::mul(a, b, 5);
+
+    REQUIRE(e == c);
+    REQUIRE(f == d);
 }
 
-TEST_CASE("Multiplication of two UnivariateSeries", "[UnivariateSeries]")
+TEST_CASE("Exponentiation of UnivariateExprPolynomial with precision", "[UnivariateSeries]")
 {
     RCP<const Symbol> x  = symbol("x");
-    RCP<const UnivariateSeries> a = univariate_series(x, 5, {{0, 1_z}, {1, 2_z}, {2, 1_z}});
-    RCP<const UnivariateSeries> b = univariate_series(x, 3, {{0, -1_z}, {1, -2_z}, {2, -1_z}});
-    RCP<const UnivariateSeries> e = univariate_series(x, 5, {{0, 1_z}, {1, 4_z}, {2, 6_z}, {3, 4_z}, {4, 1_z}});
-    RCP<const UnivariateSeries> f = univariate_series(x, 3, {{0, -1_z}, {1, -4_z}, {2, -6_z}});
+    UnivariateExprPolynomial a(univariate_polynomial(x, 2, {{0, 1}, {1, 2}, {2, 1}}));
+    UnivariateExprPolynomial b(univariate_polynomial(x, 2, {{0, -1}, {1, -2}, {2, -1}}));
+    UnivariateExprPolynomial c(univariate_polynomial(x, 3, {{0, 1}, {1, 4}, {2, 6}, {3, 4}}));
+    UnivariateExprPolynomial d(univariate_polynomial(x, 4, {{0, -1}, {1, -6}, {2, -15}, {3, -20}, {4, -15}}));
+    
+    UnivariateExprPolynomial e = UnivariateSeries::pow(a, 2, 4);
+    UnivariateExprPolynomial f = UnivariateSeries::pow(b, 3, 5);
+
+    REQUIRE(e == c);
+    REQUIRE(f == d);
+}
 
-    RCP<const UnivariateSeries> c = mul_uni_series(*a, *a);
-    RCP<const UnivariateSeries> d = mul_uni_series(*a, *b);
+TEST_CASE("Differentiation of UnivariateSeries", "[UnivariateSeries]")
+{
+    RCP<const Symbol> x  = symbol("x");
+    UnivariateExprPolynomial a(univariate_polynomial(x, 2, {{0, 1}, {1, 2}, {2, 1}}));
+    UnivariateExprPolynomial b(univariate_polynomial(x, 1, {{0, 2}, {1, 2}}));
+    UnivariateExprPolynomial c(univariate_polynomial(x, 1, {{0, 2}, {1, 2}}));
+    REQUIRE(UnivariateSeries::diff(a, b) == c);
```

Comment:
You just need this change, `UnivariateExprPolynomial b(univariate_polynomial(x, 0, {{0, 1}}));`